src/Microsoft.Framework.Logging.EventLog/EventLogLogger.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+
+namespace Microsoft.Framework.Logging.EventLog
+{
+    /// <summary>
+    /// A logger that writes messages to Windows Event Log.
+    /// </summary>
+    public class EventLogLogger : ILogger
+    {
+        private readonly System.Diagnostics.EventLog _eventLog;
+        private readonly string _name;
+        private readonly EventLogSettings _settings;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EventLogLogger"/> class.
+        /// </summary>
+        /// <param name="name">The name of the logger.</param>
+        public EventLogLogger(string name)
+            : this(name, settings: null)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EventLogLogger"/> class.
+        /// </summary>
+        /// <param name="name">The name of the logger.</param>
+        /// <param name="settings">The <see cref="EventLogSettings"/>.</param>
+        public EventLogLogger(string name, EventLogSettings settings)
+        {
+            _name = string.IsNullOrEmpty(name) ? nameof(EventLogLogger) : name;
+            _settings = settings;
+
+            var logName = string.IsNullOrEmpty(settings.LogName) ? "Application" : settings.LogName;
+            var sourceName = string.IsNullOrEmpty(settings.SourceName) ? "Application" : settings.SourceName;
+            var machineName = string.IsNullOrEmpty(settings.MachineName) ? "." : settings.MachineName;
+
+            // Due to the following reasons, we cannot have these checks either here or in IsEnabled method:
+            // 1. Log name & source name existence check only works on local computer.
+            // 2. Source name existence check requires Administrative privileges.
+
+            _eventLog = new System.Diagnostics.EventLog(logName, machineName, sourceName);
+        }
+
+        /// <inheritdoc />
+        public IDisposable BeginScopeImpl(object state)
+        {
+            return null;
```

Comment:
That's a good question, I'm not sure what the answer is. I think that maybe the main idea for this logger is because it's super easy to get to on Windows - nothing special has to be installed or set up on the machine. So maybe having *some* logging is better than none...